cli/azd/pkg/tools/kubectl/kubectl.go

```diff
@@ -0,0 +1,265 @@
+}
+
+// Sets the k8s context to use for future CLI commands
+func (cli *kubectlCli) ConfigUseContext(ctx context.Context, name string, flags *KubeCliFlags) (*exec.RunResult, error) {
+	res, err := cli.Exec(ctx, flags, "config", "use-context", name)
+	if err != nil {
+		return nil, fmt.Errorf("failed setting kubectl context: %w", err)
+	}
+
+	return &res, nil
+}
+
+// Views the current k8s configuration including available clusters, contexts & users
+func (cli *kubectlCli) ConfigView(
+	ctx context.Context,
+	merge bool,
+	flatten bool,
+	flags *KubeCliFlags,
+) (*exec.RunResult, error) {
+	kubeConfigDir, err := getKubeConfigDir()
+	if err != nil {
+		return nil, err
+	}
+
+	args := []string{"config", "view"}
+	if merge {
+		args = append(args, "--merge")
+	}
+	if flatten {
+		args = append(args, "--flatten")
+	}
+
+	runArgs := exec.NewRunArgs("kubectl", args...).
+		WithCwd(kubeConfigDir).
+		WithEnv(environ(cli.env))
+
+	res, err := cli.executeCommandWithArgs(ctx, runArgs, flags)
+	if err != nil {
+		return nil, fmt.Errorf("kubectl config view: %w", err)
+	}
+
+	return &res, nil
+}
+
+func (cli *kubectlCli) ApplyPipe(ctx context.Context, input string, flags *KubeCliFlags) (*exec.RunResult, error) {
+	runArgs := exec.
+		NewRunArgs("kubectl", "apply", "-f", "-").
+		WithEnv(environ(cli.env)).
+		WithStdIn(strings.NewReader(input))
+
+	res, err := cli.executeCommandWithArgs(ctx, runArgs, flags)
+	if err != nil {
+		return nil, fmt.Errorf("kubectl apply -f: %w", err)
+	}
+
+	return &res, nil
+}
+
+// Applies manifests from the specified input
+func (cli *kubectlCli) ApplyFiles(ctx context.Context, path string, flags *KubeCliFlags) error {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return fmt.Errorf("failed reading files in path, '%s', %w", path, err)
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		ext := filepath.Ext(entry.Name())
+		if !(ext == ".yaml" || ext == ".yml") {
+			continue
+		}
+
+		filePath := filepath.Join(path, entry.Name())
+		fileBytes, err := os.ReadFile(filePath)
+		if err != nil {
+			return fmt.Errorf("failed reading manifest file '%s', %w", filePath, err)
+		}
+
+		yaml := string(fileBytes)
+		replaced, err := envsubst.Eval(yaml, func(name string) string {
+			if val, has := cli.env[name]; has {
+				return val
+			}
+			return os.Getenv(name)
+		})
+
+		if err != nil {
+			return fmt.Errorf("failed replacing env vars, %w", err)
+		}
+
+		_, err = cli.ApplyPipe(ctx, replaced, flags)
+		if err != nil {
+			return fmt.Errorf("failed applying manifest, %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Creates a new generic secret from the specified secret pairs
+func (cli *kubectlCli) CreateSecretGenericFromLiterals(
+	ctx context.Context,
+	name string,
+	secrets []string,
+	flags *KubeCliFlags,
+) (*exec.RunResult, error) {
+	args := []string{"create", "secret", "generic", name}
+	for _, secret := range secrets {
+		args = append(args, fmt.Sprintf("--from-literal=%s", secret))
```

Comment:
Can you update `redactSensitiveData` in `command_runner.go` so that we don't show the secret values in logs?

Reply:
Also, is it possible to pass the secret value via stdin instead of on the command line (where it's easier to observe via things like procfs)?

Reply:
Will look into this.

Reply:
Updated redaction but not able to set via `stdin` in this case :(